syntax = "proto3";

message UploadFileRequest {
    string file_name = 1;
    string type = 2;
    bytes file_data = 3;

}

message UploadFileResponse {
}

message DownloadFileRequest {
  string file_name = 1;
}

message DownloadFileResponse {
    bytes file_data = 1;
}

message DeleteFileRequest {
    string file_name = 1;
}

message DeleteFileResponse {
}

service FileCloudAccess {
    rpc UploadFile(stream UploadFileRequest) returns (UploadFileResponse);
    rpc DownloadFile(DownloadFileRequest) returns (stream DownloadFileResponse);
    rpc DeleteFile(DeleteFileRequest) returns (DeleteFileResponse);
}
